src/dispatching/repls/repl.rs

```diff
@@ -53,44 +44,34 @@ where
 /// [`repl`]: crate::dispatching::repls::repl()
 /// [`UpdateListener`]: crate::dispatching::update_listeners::UpdateListener
 #[cfg(feature = "ctrlc_handler")]
-pub async fn repl_with_listener<'a, R, H, Fut, E, L, ListenerE>(
-    requester: R,
-    handler: H,
-    listener: L,
-) where
-    H: Fn(UpdateWithCx<R, Message>) -> Fut + Send + Sync + 'static,
-    Fut: Future<Output = Result<(), E>> + Send + 'static,
+pub async fn repl_with_listener<'a, R, H, E, L, ListenerE, Args>(bot: R, handler: H, listener: L)
+where
+    H: Injectable<DependencyMap, Result<(), E>, Args> + Send + Sync + 'static,
     L: UpdateListener<ListenerE> + Send + 'a,
     ListenerE: Debug,
     Result<(), E>: OnError<E>,
-    E: Debug + Send,
-    R: Requester + Clone + Send + 'static,
+    E: Debug + Send + Sync + 'static,
+    R: Requester + Clone + Send + Sync + 'static,
 {
-    let handler = Arc::new(handler);
+    use crate::dispatching::Dispatcher;
+
+    // Other update types are of no interest to use since this REPL is only for
+    // messages. See <https://github.com/teloxide/teloxide/issues/557>.
+    let ignore_update = |_upd| Box::pin(async {});
+
+    #[allow(unused_mut)]
```

Comment:
why is this needed?

Reply:
This is because of the `#[cfg(feature = "ctrlc_handler")]` below. But I've just noticed that we already have it on the whole `repl_with_listener`, so I think it should be removed.